cmd/kubeadm/app/phases/etcd/local.go

```diff
@@ -204,8 +203,8 @@ func GetEtcdPodSpec(cfg *kubeadmapi.ClusterConfiguration, endpoint *kubeadmapi.A
 					v1.ResourceMemory: resource.MustParse("100Mi"),
 				},
 			},
-			LivenessProbe: staticpodutil.LivenessProbe(probeHostname, etcdHealthEndpoint, probePort, probeScheme),
-			StartupProbe:  staticpodutil.StartupProbe(probeHostname, etcdHealthEndpoint, probePort, probeScheme, cfg.APIServer.TimeoutForControlPlane),
+			LivenessProbe: staticpodutil.LivenessProbe(probeHostname, "/health?serializable=true", probePort, probeScheme),
```

Comment:
```suggestion
			LivenessProbe: staticpodutil.LivenessProbe(probeHostname, "/health?exclude=NOSPACE&serializable=true", probePort, probeScheme),
```

Reply:
I don't think we need '\' here. The library can handle it automatically, no need to worry about it. 

Please try the URL in browser, and browser can handle it, just in the same way as the golang library does.

Reply:
When I test it, I find there is a thing. etcd will not be killed after the first kill.

In the past, if there is NOSPACE alert, then pod's liveness will fail for /health.
Then etcd pod will be started again. The startup probe will never pass. So no liveness probe will be triggered.

Pod will keep running like below
```
[root@paco ~]# kubectl get events  -n kube-system   -w| grep etcd
13m         Warning   Unhealthy        pod/etcd-paco                   Startup probe failed: HTTP probe failed with statuscode: 503
12m         Normal    Killing          pod/etcd-paco                   Stopping container etcd

[root@paco ~]# kubectl get pod  -n kube-system  etcd-paco
NAME        READY   STATUS    RESTARTS   AGE
etcd-paco   1/1     Running   0          13m
```

Reply:
So the difference
- adding `exclude=NOSPACE`: no restart if it is nospace
- without `exclude=NOSPACE`: one restart

Reply:
It's expected behavior for `/health?exclude=NOSPACE&serializable=true`.  It depends on what's your expected behavior. 

# With `exclude=NOSPACE`

- Good side: Users can still serve range/read requests. Administrator can fix the space issue for each member one by one, and the etcd cluster can always serve range/read requests during the process.
- Bad side: The member (even the whole cluster) is actually in unhealthy status. Users may not notice it unless there is an automatic monitoring system, such as prometheus. 

# Without `exclude=NOSPACE`
- Good side: The etcd member will be restarted by the liveness Probe, and will never get started due to startup Probe. Accordingly it can get immediate attention from administrator;
- Bad side: It can't serve even range/read requests. The system which depends on the etcd cluster will be totally down.

# Proposal
How about expose the configuration to users? And the default value is with `exclude=NOSPACE` ?